fuzz-afl/reproducers/reproduce_pnm.rs

```diff
@@ -0,0 +1,25 @@
+extern crate image;
+
+use image::ImageDecoder;
+use image::error::{ImageError, ImageResult, LimitError, LimitErrorKind};
+
+mod utils;
+
+#[inline(always)]
+fn pnm_decode(data: &[u8]) -> ImageResult<Vec<u8>> {
+    let decoder = image::pnm::PnmDecoder::new(data)?;
+    let (width, height) = decoder.dimensions();
+
+    if width.saturating_mul(height) > 4_000_000 {
+        return Err(ImageError::Limits(LimitError::from_kind(LimitErrorKind::DimensionError)));
+    }
+
+    let mut buf = vec![];
+    decoder.read_image(&mut buf)?;
```

Comment:
This is the panic you observe, it's a bug in the fuzzer script and not in `image`. According to [that functions  specification](https://docs.rs/image/0.23.0/image/trait.ImageDecoder.html#method.read_image), you should provide a buffer large enough to hold the full output image and else the decoder panics. 

An `ImageError::ParameterError` is not returned because the precondition can and should be publicly checked (although we *could* change it to instead return an error). But the reason is to make this misuse obvious: fuzzing only becomes meaningful with an actual buffer because without one there will never be any decoding executed.

This should instead probably read: 

```rust
let mut buf = vec![0; decoder.total_bytes()];
decoder.read_image(&mut buf);
```

or even

```rust
DynamicImage::from_decoder(decoder)
```

Reply:
Ah, thanks! I already suspected something like that.

Reply:
Yep, now all inputs pass the reproducer test :slightly_smiling_face: